### ISSUE TITLE
Generate oauth authorization code not working

### ISSUE THREAD

**Issue by danschwarz** [pixelfed/pixelfed#4293]
I'm experiencing this defect when trying to use the Pixelfed 0.11.5 instance [pixelfed.de](https://pixelfed.de).
Trying to login using the [toot](https://github.com/ihabunek/toot) client requires an oauth authorization code. I'm directed to the browser URL to authorize toot. 

![image](https://user-images.githubusercontent.com/3261094/231031089-71c356c9-df4d-45b9-8613-9e7f05ffb5f6.png)

Unfortunately, clicking "Authorize" does nothing.  I would expect to see a generated authorization code.

This might be an issue unique to the pixelfed.de server, but I'm posting here in case it's a more generalized issue with 0.11.5.  Thanks!

**Comment by mitexleo** [comment]
You shouldn't see a generated authorized code. You should be redirected to the app.

**Comment by danschwarz** [comment]
I'm not redirected anywhere, the authorize button does nothing.

**Comment by mitexleo** [comment]
> I'm not redirected anywhere, the authorize button does nothing.

That's weird

**Comment by horazont** [comment]
> You shouldn't see a generated authorized code. You should be redirected to the app.

This is not correct, because toot [uses the "magic" `urn:ietf:wg:oauth:2.0:oob` redirect URI](https://github.com/ihabunek/toot/blob/master/toot/api.py#L80-L90) which should cause Pixelfed to *display* the token instead of issuing a redirect.

EDIT: And for the record, the behaviour is still the same with 0.12.1 (local test setup using docker w/ oauth enabled) and 0.12.3 (pixelfed.de).

**Comment by rjp** [comment]
Toot PR to fudge this for any instances with similar issues - https://github.com/ihabunek/toot/pull/512
